tuxemon/technique/effects/run.py

```diff
@@ -37,26 +37,36 @@ def apply(
         player = user.owner
         assert combat and player
 
-        def escape(level_user: int, level_target: int, attempts: int) -> bool:
-            escaping: bool = False
-            escape = 0.4 + (0.15 * (attempts + level_user - level_target))
-            if random.random() <= escape:
-                escaping = True
-            return escaping
+        game_variables = player.game_variables
+        attempts = game_variables.get("run_attempts", 0)
+        escape_method = game_variables.get("method_escape", "default")
+        escape_ai_method = game_variables.get("method_escape_ai", "default")
 
-        var = player.game_variables
-        if "run_attempts" not in var:
-            var["run_attempts"] = 0
-        # monster in the player party
+        # Check if the user is in the player party or NPC party
         if user in combat.monsters_in_play_right:
-            if escape(user.level, target.level, var["run_attempts"]):
-                var["run_attempts"] += 1
-                ran = True
-        # monster in the NPC party
+            escape_method = escape_method
         elif user in combat.monsters_in_play_left:
+            escape_method = escape_ai_method
+        else:
+            return {
+                "success": True,
+                "damage": 0,
+                "element_multiplier": 0.0,
+                "should_tackle": False,
+                "extra": None,
+            }
+
+        # Attempt to escape
+        if formula.attempt_escape(
+            escape_method,
+            user,
+            target,
+            attempts,
+        ):
+            game_variables["run_attempts"] = attempts + 1
```

Comment:
Doesn't this mean that every time we successfully run away from a battle, running away becomes permanently easier? Seems a bit odd. Is this intentional?

Reply:
what if it wasn't my choice? https://github.com/Tuxemon/Tuxemon/pull/1395#issuecomment-1279959092

Reply:
come on, let's tag the formula's creator @Sanglorian

Reply:
@kerizane I realized that the attempt counter might not be necessary, as the final decision is determined by the formula in `formula.py`, which returns a boolean value. This means that someone could develop a formula without considering the attempts counter.

Reply:
added other methods (no use of attempts)